src/app/_services/editor-pref.service.spec.ts

```diff
@@ -0,0 +1,62 @@
+import { QueryList } from '@angular/core';
+import { async, TestBed } from '@angular/core/testing';
+import { CodemirrorComponent } from 'ng2-codemirror';
+
+import { EditorPrefService } from '.';
+
+function makeQueryList(): QueryList<CodemirrorComponent> {
+  // tslint:disable: no-any
+  return ([
+    {
+      instance: {
+        setOption(name: string, value: string): void {
+          console.log('set ' + name + ' to ' + value);
```

Comment:
don't use console.log. if you want to check if a function is called, use a spy, for example:

    instance: {
        setOption: jasmine.createSpy('setOption'),
    }

...and later check if it has been called:

    expect(queryList[0].instance.setOptions).toHaveBeenCalledWith('theme', 'midnight');